test/controllers/update/update.beneficial.owner.gov.spec.ts

```diff
@@ -481,64 +430,234 @@ describe("UPDATE BENEFICIAL OWNER GOV controller", () => {
       beneficialOwnerGov["start_date-month"] = inTheFuture.month.toString();
       beneficialOwnerGov["start_date-year"] = inTheFuture.year.toString();
       const resp = await request(app)
         .post(UPDATE_BENEFICIAL_OWNER_GOV_URL)
         .send(beneficialOwnerGov);
+
+      assertOnlyDayNotInPastErrors(resp);
+    });
+
+    test(`renders the ${UPDATE_BENEFICIAL_OWNER_GOV_PAGE} page without date errors including DATE_NOT_IN_PAST_OR_TODAY error when start date is today`, async () => {
+      const beneficialOwnerGov = { ...REQ_BODY_BENEFICIAL_OWNER_GOV_FOR_START_DATE_VALIDATION };
+      const today = DateTime.now();
+      beneficialOwnerGov["start_date-day"] = today.day.toString();
+      beneficialOwnerGov["start_date-month"] = today.month.toString();
+      beneficialOwnerGov["start_date-year"] = today.year.toString();
+      const resp = await request(app)
+        .post(UPDATE_BENEFICIAL_OWNER_GOV_URL)
+        .send(beneficialOwnerGov);
+
       expect(resp.status).toEqual(200);
       expect(resp.text).toContain(BENEFICIAL_OWNER_GOV_PAGE_HEADING);
       expect(resp.text).not.toContain(ErrorMessages.ENTER_DATE);
       expect(resp.text).not.toContain(ErrorMessages.DAY);
       expect(resp.text).not.toContain(ErrorMessages.MONTH);
       expect(resp.text).not.toContain(ErrorMessages.YEAR);
       expect(resp.text).not.toContain(ErrorMessages.INVALID_DATE);
-      expect(resp.text).toContain(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY);
+      expect(resp.text).not.toContain(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY);
       expect(mockSaveAndContinue).not.toHaveBeenCalled();
     });
 
-    test(`renders the ${UPDATE_BENEFICIAL_OWNER_GOV_PAGE} page without date errors including DATE_NOT_IN_PAST_OR_TODAY error when start date is today`, async () => {
-      const beneficialOwnerGov = { ...REQ_BODY_BENEFICIAL_OWNER_GOV_FOR_DATE_VALIDATION };
-      const today = DateTime.now();
-      beneficialOwnerGov["start_date-day"] = today.day.toString();
-      beneficialOwnerGov["start_date-month"] = today.month.toString();
-      beneficialOwnerGov["start_date-year"] = today.year.toString();
+    test(`renders the ${UPDATE_BENEFICIAL_OWNER_GOV_PAGE} page with only INVALID_DATE error when ceased date month and year are empty`, async () => {
+      const beneficialOwnerGov = { ...UPDATE_BENEFICIAL_OWNER_GOV_MOCK_FOR_CEASE_VALIDATION };
+      beneficialOwnerGov["ceased_date-day"] = "01";
+      beneficialOwnerGov["ceased_date-month"] = "";
+      beneficialOwnerGov["ceased_date-year"] = "";
+
+      const resp = await request(app)
+        .post(UPDATE_BENEFICIAL_OWNER_GOV_URL)
+        .send(beneficialOwnerGov);
+
+      assertEmptyMonthAndYearErrors(resp);
+    });
+
+    test(`renders the ${UPDATE_BENEFICIAL_OWNER_GOV_PAGE} page with only INVALID_DATE error when ceased date day and year are empty`, async () => {
+      const beneficialOwnerGov = { ...UPDATE_BENEFICIAL_OWNER_GOV_MOCK_FOR_CEASE_VALIDATION };
```

Comment:
not sure you needed to copy the mock - just add to one without ceased date & cast to any.

Reply:
I personally prefer the approach we've taken so will leave it unless you feel strongly :)